src/include/cpp-btree/btree.h

```diff
@@ -1207,21 +1139,27 @@ class btree {
   void swap(btree &x);
 
   const key_compare &key_comp() const noexcept {
-    return root_.template get<0>();
+    return *static_cast<const key_compare*>(&root_);
   }
   template <typename K, typename LK>
   bool compare_keys(const K &x, const LK &y) const {
-    return compare_internal::compare_result_as_less_than(key_comp()(x, y));
+    return compare_result_as_less_than(key_comp()(x, y));
   }
 
-  value_compare value_comp() const { return value_compare(key_comp()); }
+  // Dump the btree to the specified ostream. Requires that operator<< is
+  // defined for Key and Value.
+  void dump(std::ostream &os) const {
```

Comment:
That's a Ceph-specific extension, I bet.

Reply:
yes. i intend to keep it in another PR. as this change is supposed to be backward compatible from the feature set perspective.

Reply:
My guess is that we will need tit sooner or later for debugging seastore. Let's keep.